pkg/cmd/server/start/start_api.go

```diff
@@ -92,8 +92,10 @@ func NewCommandStartMasterAPI(name, basename string, out io.Writer) (*cobra.Comm
 		config.EtcdConfig = nil
 		if config.KubernetesMasterConfig != nil {
 			if masterAddr.Provided {
-				glog.V(2).Infof("Using a master address override %q", masterAddr.Host)
-				config.KubernetesMasterConfig.MasterIP = masterAddr.Host
+				if ip := net.ParseIP(masterAddr.Host); ip != nil {
```

Comment:
@smarterclayton would you rather conditionally set it here, or let it get in and fail validation? calling `OverrideConfig` only in the non-write-config case fixes the bug just as well